drivers/overlay/encryption.go

```diff
@@ -43,17 +45,39 @@ func (k *key) String() string {
 	return ""
 }
 
+// hosts-pair spis
 type spi struct {
 	forward int
 	reverse int
 }
 
-func (s *spi) String() string {
+func (s spi) String() string {
 	return fmt.Sprintf("SPI(FWD: 0x%x, REV: 0x%x)", uint32(s.forward), uint32(s.reverse))
 }
 
+// hosts-pair keys
+type hpk struct {
+	forward []byte
+	reverse []byte
+}
+
+func (h hpk) String() string {
+	return fmt.Sprintf("Key(FWD: %s, REV: %s)",
+		hex.EncodeToString(h.forward)[0:5], hex.EncodeToString(h.reverse)[0:5])
+}
+
+// hosts-pair encryption data
+type encrElem struct {
+	spis spi
+	keys hpk
+}
+
+func (ee *encrElem) String() string {
```

Comment:
If we should not ever print them even a truncated key in the debug level logs, then better remove this.
My assumption was that only an admin can access the debug logs, still the full key wouldn't be printed.